maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java

```diff
@@ -49,7 +49,9 @@
  * assist those clients that employ the repository system outside of an IoC container, Maven plugins should instead
  * always use regular dependency injection to acquire the repository system.
  *
+ * @deprecated See {@link MavenSessionBuilderSupplier}
  */
+@Deprecated
 public final class MavenRepositorySystemUtils {
```

Comment:
Shouldn't this class be re-implemented on top of MavenSessionBuilderSupplier, so that any change in the new one can be somewhat reflected in the old one ?